orttraining/orttraining/python/training/ortmodule/_logger.py

```diff
@@ -23,31 +26,71 @@ class LogLevel(IntEnum):
 
 
 @contextmanager
-def suppress_os_stream_output(suppress_stdout=True, suppress_stderr=True, log_level=LogLevel.WARNING):
-    """Suppress output from being printed to stdout and stderr if log_level is WARNING or higher.
+def suppress_os_stream_output(on_exit: Optional[Callable] = None):
+    """Suppress output from being printed to stdout and stderr.
 
-    If there is any output detected, a single warning is issued in the context
+    If on_exit is not None, it will be called when the context manager exits.
     """
 
     # stdout and stderr is written to a tempfile instead
-    stdout = sys.stdout
-    stderr = sys.stderr
+    _, fo_file_path = tempfile.mkstemp()
```

Comment:
If I am not mistaken, the temporary file created this way must be marked for deletion after work is completed. Shall we instead use the context manager?

Reply:
good catch, man!